packages/swapper/src/swappers/ZrxSwapper/utils/fetchFromZrx.ts

```diff
@@ -71,3 +83,51 @@ export const fetchFromZrx = async <T extends 'price' | 'quote'>({
 
   return Ok(maybeZrxPriceResponse.unwrap().data)
 }
+
+export const fetchFromZrxPermit2 = async <T extends 'price' | 'quote'>({
```

Comment:
We don't ever consume this with `price`.
Two qs: 
1. Is refetching an actual quote in `getUnsignedEvmTransaction` something we want, and can even do given the ramifications of how permit2 flow works?
2. Do we want to only get a rate quote in `getZrxTradeQuote`, or same thing, that won't work given how permit2 works?

Reply:
Gm:
1. Refetching is only done in `getUnsignedEvmTransaction` when `transactionMetadata` is absent from the quote, indicating the quote is from the V1 api. In this case, we'd need to fetch a fresh quote from the V1 api to obtain the transaction metadata. This will be removed once V2 api has been stable in prod for ~2 weeks.
2. When V1 api is removed, we can refactor to simplify everything. V2 flow uses only a single fetch of a `quote`, as opposed to the v1 flow which needs a `price` and then a `quote`.

TL;DR good observation and we'll clean up when we are confident we dont want to toggle back to the V2 api.